lumicks/pylake/force_calibration/power_spectrum.py

```diff
@@ -22,7 +46,60 @@ class PowerSpectrum:
         The total duration of the original data. [seconds]
     """
 
-    def __init__(self, data, sample_rate, unit="V", window_seconds=None):
+    def __init__(
+        self,
+        frequency,
+        power,
+        sample_rate,
+        total_duration,
+        unit="V",
+        *,
+        num_points_per_block=1,
+        total_samples_used=None,
+        variance=None,
+    ):
+        """Power spectrum
+
+        frequency : array_like
+            Frequency axis
+        power : array_like
+            Power spectral values
+        sample_rate : int
+            Sample rate
+        total_duration : float
+            Total measurement duration
+        unit : str
+            Unit of the spectrum
+        num_points_per_block : int
+            Number of points per block of the source data
+        total_samples_used : int
+            Total samples used to compute FFT
+        variance : array_like
+            Variance of each power spectral density point
+        """
+        self._frequency = np.asarray(frequency)
+        self._power = np.asarray(power)
+        self.sample_rate = sample_rate
+        self.total_duration = total_duration
+        self.unit = unit
+
+        self.total_samples_used = total_samples_used
+        self._raw_variance = variance
+        self._src_num_points_per_block = num_points_per_block
```

Comment:
since this is private anyway, what's the `src` prefix for/mean?

Reply:
It stands for source and was a reminder that that refers to the num points per block of the source material (pre blocking). That said, it is probably superfluous, since you can so clearly see it in the property already.